hydra-node/src/Hydra/Chain/ScriptRegistry.hs

```diff
@@ -105,54 +102,56 @@ publishHydraScripts networkId socketPath sk = do
  where
   vk = getVerificationKey sk
 
+-- | Exception raised when building the script publishing transactions.
+data PublishScriptException
+  = FailedToBuildPublishingTx (TxBodyErrorAutoBalance Era)
+  | FailedToFindUTxOToCoverDeposit {totalDeposit :: Coin}
+  deriving (Show)
+  deriving anyclass (Exception)
+
+-- | Builds a chain of script publishing transactions.
+-- Throws: PublishScriptException
 buildScriptPublishingTxs ::
```

Comment:
@ffakenz @v0d1ch You might want to review this new implementation as you both have worked on it recently.